copy/single.go

```diff
@@ -466,14 +482,18 @@ func (ic *imageCopier) copyLayers(ctx context.Context) error {
 		// A call to copyGroup.Wait() is done at this point by the defer above.
 		return nil
 	}(); err != nil {
-		return err
+		return nil, err
 	}
 
+	compressionAlgos := set.New[string]()
 	destInfos := make([]types.BlobInfo, numLayers)
 	diffIDs := make([]digest.Digest, numLayers)
 	for i, cld := range data {
+		if cld.destInfo.CompressionAlgorithm != nil {
+			compressionAlgos.Add(cld.destInfo.CompressionAlgorithm.Name())
+		}
 		if cld.err != nil {
```

Comment:
Oops, I apologize for missing this earlier … `cld.destInfo` is only valid if `cld.err` == nil`; so the new code should be added below.

Reply:
done.